### PR TITLE
PMM-7123 add external exporter

### PR DESCRIPTION
**Pull request by lunaticusgreen**
[PMM-7123](https://jira.percona.com/browse/PMM-7123) [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import { LinkTooltip } from 'shared/components/Elements/LinkTooltip/LinkTooltip';
+import { Button, useTheme } from '@grafana/ui';
+import { Field } from 'react-final-form';
+import { RadioButtonGroupAdapter } from 'shared/components/Form/FieldAdapters/FieldAdapters';
+import { ExternalServiceConnectionDetailsFormPartProps } from '../FormParts.types';
+import { getStyles } from '../FormParts.styles';
+import { Messages } from '../FormParts.messages';
+import { schemaOptions } from '../FormParts.constants';
+
+export const ExternalServiceConnectionDetails: FC<ExternalServiceConnectionDetailsFormPartProps> = ({
```
Comment (tiagomotasantos) [inline pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx:13]: Can we afford tests for this component?

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    } catch (e) {
+      console.log(e);
+    }
+
+    return value;
+  };
+
+  return (
+    <div className={styles.groupWrapper}>
+      <h4 className={styles.sectionHeader}>External service connection details</h4>
```
Comment (tiagomotasantos) [inline pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx:41]: Some strings need to go to Messages file.

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx (hunk 2 above)
Comment (lunaticusgreen) [reply pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx:41]: done, thanks

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.tsx [hunk 3]
```diff
@@ -46,33 +46,52 @@ const AddRemoteInstance: FC<AddRemoteInstanceProps> = ({
...
+            remoteInstanceCredentials={remoteInstanceCredentials}
+            loading={loading}
+            instanceType={type}
+            form={form}
+          />
+        )}
       </>
     ),
-    [],
+    [type],
```
[the remark this answers is not in the document]
Comment (lunaticusgreen) [reply pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.tsx:71]: Actually, we don't need to have a type here too, thanks

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.test.tsx [hunk 4]
```diff
@@ -5,11 +5,19 @@ import AddRemoteInstance from './AddRemoteInstance';
...
 
     root.find('form[data-qa="add-remote-instance-form"]').simulate('submit');
   });
+
+  it('should render for external', () => {
+    const type = 'external';
+
+    const root = mount(<AddRemoteInstance instance={{ type, credentials: {} }} selectInstance={jest.fn()} />);
+
+    root.find('form[data-qa="add-remote-instance-form"]').simulate('submit');
```
[the remark this answers is not in the document]
Comment (lunaticusgreen) [reply pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.test.tsx:21]: replaced

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.test.tsx [hunk 5]
```diff
@@ -5,11 +5,19 @@ import AddRemoteInstance from './AddRemoteInstance';
...
 
     const root = mount(<AddRemoteInstance instance={{ type, credentials: {} }} selectInstance={jest.fn()} />);
 
     root.find('form[data-qa="add-remote-instance-form"]').simulate('submit');
   });
+
+  it('should render for external', () => {
+    const type = 'external';
+
+    const root = mount(<AddRemoteInstance instance={{ type, credentials: {} }} selectInstance={jest.fn()} />);
```
[the remark this answers is not in the document]
Comment (lunaticusgreen) [reply pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.test.tsx:28]: Good point, fixed

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.service.tsx [hunk 6]
```diff
@@ -98,3 +104,40 @@ export const toPayload = (values, discoverName) => {
 
   return data;
 };
+
+export const toExternalServicePayload = (values) => {
```
[the remark this answers is not in the document]
Comment (lunaticusgreen) [reply pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.service.tsx]: done

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx (hunk 1 above)
Comment (lunaticusgreen) [reply pmm-app/src/pmm-add-instance/components/AddRemoteInstance/FormParts/ExternalServiceConnectionDetails/ExternalServiceConnectionDetails.tsx:13]: Good point, fixed

File: pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.service.tsx [hunk 7]
```diff
@@ -101,11 +101,12 @@ export const toPayload = (values, discoverName?) => {
   }
 
   data.metrics_mode = 1;
+  console.log(values, data);
```
Comment (tiagomotasantos) [inline pmm-app/src/pmm-add-instance/components/AddRemoteInstance/AddRemoteInstance.service.tsx:104]: Just remove console.log (below too) and its good to go!

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]

**Review by tiagomotasantos — APPROVED** [review]

**Pull request merged**